spec/suites/layer/vector/RectangleSpec.js

```diff
@@ -0,0 +1,144 @@
+describe('Rectangle', function () {
+	var map, container;
+
+	beforeEach(function () {
+		container = createContainer();
+		map = L.map(container, {center: [55.8, 37.6], zoom: 6});
+	});
+
+	afterEach(function () {
+		removeMapContainer(map, container);
+	});
+
+	describe("#initialize", function () {
+		it("should never be flat", function () {
+			var latLngs = [[1, 2], [3, 4]];
+
+			var rectangle = L.rectangle(latLngs);
+
+			expect(L.LineUtil.isFlat(rectangle._latlngs)).to.be(false);
+			expect(rectangle.getLatLngs()).to.eql(rectangle._latlngs);
+		});
+
+		it("doesn't overwrite the given latlng array", function () {
+			var originalLatLngs = [
+				[1, 2],
+				[3, 4]
+			];
+			var sourceLatLngs = originalLatLngs.slice();
+
+			var rectangle = L.rectangle(sourceLatLngs);
+
+			expect(sourceLatLngs).to.eql(originalLatLngs);
+			expect(rectangle._latlngs).to.not.eql(sourceLatLngs);
+		});
+
+		it("cannot be called with an empty array", function () {
+			// Throws error due to undefined lat
+			expect(L.rectangle).withArgs([]).to.throwException();
```

Comment:
```suggestion
			expect(L.rectangle([])).to.throwException();
```

Reply:
I tried to do this previously but got the following error message:

```
Chrome Headless 99.0.4844.84 (Windows 10) Rectangle #initialize cannot be called with an empty array FAILED
        TypeError: Cannot read properties of undefined (reading 'lat')
            at LatLngBounds.getNorth (src/Leaflet.js:1260:28)
            at LatLngBounds.getNorthWest (src/Leaflet.js:1230:28)
            at NewClass._boundsToLatLngs (src/Leaflet.js:13092:19)
            at NewClass.initialize (src/Leaflet.js:13079:50)
            at new NewClass (src/Leaflet.js:297:22)
            at Object.rectangle (src/Leaflet.js:13102:11)
            at Context.<anonymous> (spec/suites/layer/vector/RectangleSpec.js:38:13)
```
Any suggestions on how to fix this?

Reply:
```
expect(function () {
    L.rectangle([]);
}).to.throwException();
```